docs/products/opensearch/howto/opensearch-search-and-python.rst

```diff
@@ -0,0 +1,471 @@
+
+.. code:: python
+
+    import os
+
+    from dotenv import load_dotenv
+    from opensearchpy import OpenSearch
+
+
+    load_dotenv()
+    INDEX_NAME = "epicurious-recipes"
+    SERVICE_URI = os.getenv("SERVICE_URI")
+    client = OpenSearch(SERVICE_URI, use_ssl=True)
+
+
+.. tip::
+
+    The ``SERVICE_URI`` value can be found in the Aiven Console dashboard.
+
+Upload data to OpenSearch using Python
+--------------------------------------
+
+Once you're connected, the next step should be to inject data into our cluster. Follow the instructions in how to :ref:`load the sample data <load-data-with-python>` into your OpenSearch cluster. 
+
+You can run from the demo following::
+
+  python index.py send-data
+
+Once the data is sent, we can :ref:`retrieve the data mapping <get-mapping-with-python>` to explore the structure of the data, with their respective fields and types. 
+
+From the `demo <https://github.com/aiven/demo-opensearch-python/blob/main/index.py#L50>`_, you can check the data mapping running::
+
+  python index.py get-mapping
+
+
+All set to start writing your own search queries.
+
+Query the data
+''''''''''''''
+
+``search()`` method
+-------------------
+
+We have an OpenSearch client and data in your cluster, so we can start writing search queries. The method ``search()`` provided by the Python OpenSearch client library will be used to perform the queries. 
+
+Find below the method signature's::
+
+  client.search: (body=None, index=None, doc_type=None, params=None, headers=None)
+
+All the parameters are optional in the ``search()`` method. We will define Python ``dict()`` objects for the ``body`` parameter and use. Here is an example:
+
+.. code:: python
+   query_body = {
+                  "query": {
+                    "match_all": {}
+                  }
+                }
+
+
+The parameters that we will be using to configure our search queries are the ``index`` and ``body``. The ``index`` will not be changing, but we will be modifying the ``body`` parameter value according to the query purpose.
+
+Search lite API and query DSL
+-----------------------------
+There are two ways of performing search queries in OpenSearch: `Search Lite API <https://opensearch.org/docs/1.2/opensearch/rest-api/search/>`_ and `OpenSearch query domain-specific language (DSL) <https://opensearch.org/docs/latest/opensearch/query-dsl/index/>`_.
+In the Search Lite API, it is used the ``q`` parameter to run a query parameter search. But those kinds of queries do not support the Query DSL, and they work better for simple searches because it is expected a string as a parameter, for example:
+
+.. code-block:: python
+
+    client.search({
+        index: 'recipes',
+        q: 'ingredients:broccoli AND calories:(>=100 AND <200)'
+    })
+
+
+With Query DSL, the field ``body`` expects a dictionary object which makes it easier to construct more complex search queries. As an example:
+
+.. code-block:: python
+
+     query_body = {
```

Comment:
would you like to add command line shortcuts for all other examples, like you did above for loading data and getting schema?

Reply:
Loading data and getting schemas are explained in different articles in more details, so it is okay to have some easy commands.

The article is about writing search queries, so I think it should be bring the queries. I am not sure if the format I chose it is the best one, but still I think it is important to be very explicitly about the queries.